fvm/executionParameters.go

```diff
@@ -336,3 +357,39 @@ func GetExecutionMemoryLimit(
 
 	return uint64(memoryLimitRaw), nil
 }
+
+func GetMinimumExecutionVersion(
+	log zerolog.Logger,
+	ctx Context,
+	env environment.Environment,
+) (semver.Version, error) {
+	if !ctx.ReadVersionFromNodeVersionBeacon {
+		return semver.Version{}, nil
+	}
+
+	value, err := env.GetCurrentVersionBoundary()
+
+	if err != nil {
+		return semver.Version{}, fmt.Errorf("could not get current version boundary: %w", err)
+	}
+
+	boundary, err := convert.VersionBoundary(value)
+
+	if err != nil {
+		return semver.Version{}, fmt.Errorf("could not parse current version boundary: %w", err)
+	}
+
+	semVer, err := semver.NewVersion(boundary.Version)
```

Comment:
I feel the name "current" in `GetCurrentVersionBoundary` is confusing, because there are two cases at least:

1. boundary.Version > the version the node is currently running at.
This means there is a up coming version change at the future height, which means in this case boundary.Height is greater than the height of the block to be executed
2. boundary.Version <= the version the node is currently running at. 
In this case, there could be no version beacon was set in the past, or if set, the boundary.Height must be lower than the Height of the block to be executed.

However, we just return the `Version` value here without the Height, it means we won't distinguish these two cases. Is that right?

I think we might need two variables:
1. boundary.Version
2. IsVersionForUnreachedFutureHeight bool

`IsVersionForUnreachedFutureHeight` is true if the version is a higher version for the future height.
`IsVersionForUnreachedFutureHeight` is false if the version is a lower version for the past height or same height as the block to be executed.

I think we only need to map version when `IsVersionForUnreachedFutureHeight` is true, right?

And since there are two cases here, regardless I think it's better explain each case in comments for the map version function .

Reply:
that is why current is in the name, because it is the current version boundary that is in effect (boundary.Height <= current  height). so there is just one case, and that is `boundary.Version <= the version the node is currently running at`. The reason is that is that if we were in the situation where `boundary.Version > the version` the execution of this transaction would not be permitted (that logic is in stop control)

I already added extra comments to GetCurrentVersionBoundary but I'll add some more here.